api/src/repositories/UserRepository.ts

```diff
@@ -0,0 +1,53 @@
+import Repository from "./Repository";
+import { User } from "../models/User";
+import { Connection } from "typeorm/connection/Connection";
+import { Repository as PGRepository } from "typeorm/repository/Repository";
+import { UserDao } from "./entities/UserDao";
+
+abstract class UserRepository implements Repository<string, User> {
+  abstract save(user: User): Promise<User>;
+  abstract get(email: string): Promise<User | undefined>;
+  abstract delete(email: string): Promise<boolean>;
+  abstract findOne(query: object): Promise<User | undefined>;
+  abstract findAll(query: object): Promise<Array<User>>;
+}
```

Comment:
Are we going to have one of these for each model we have? Don't all models require these methods anyway? Maybe you could make this generic